### ISSUE TITLE
Completion op should return namespace-not-found when appropriate

### ISSUE THREAD

**Issue by eerohele** [nrepl/nrepl#230]
Given a client that sends an op like this:

```clojure
{"id" "1" "op" "completions" "prefix" "a" "ns" "my.unknown.ns"}
```

If the nREPL server doesn't know `my.unknown.ns`, it returns:

```clojure
{"id" "1" "status" ["completion-error" "done"]}
```

It would be useful if the server returned something like this instead:

```clojure
{"id" "1" "status" ["namespace-not-found" "done"]}
```

That way, clients could e.g. evaluate the `ns` form in the current view and then retry the completion op.

Currently, clients have no way of distinguishing missing namespace errors from other types of completion errors.

**Comment by bbatsov** [comment]
Yeah, that's a good point and would be easy to fix this. I'm guessing something like:

```
{"id" "1" "status" ["completion-error" "namespace-not-found" "done"]}
```

would be even better. The status is a set and we can have as many statuses as we want there.

**Comment by eerohele** [comment]
Same thing for the lookup op.

**Comment by bbatsov** [comment]
Fixed in #238.
